### PR TITLE
FCS and Scopes with no compat pbo required

### PR DESCRIPTION
**Pull request by esteldunedain**
**Experimental**

**When merged this pull request will:**
- Add a correction to ACE_FCS to account for vanilla zeroing
- Allow ACE_FCS to work without requiring a compatibility PBO
- Add a correction to ACE_Scopes to account for vanilla zeroing
- Allow ACE_Scopes to work without requiring a compatibility PBO

**Rationale**
There are currently 4 ACE systems that require "ACE-Exclusive" compatibility PBOs (that cannot be included in the content without breaking it for vanilla):
- FCS (`discreteDistance[] = {}`)
- Scopes (`discreteDistance[] = {}`)
- Explosives
- Disposable

~~The technique applied in this PR could also be applied to scopes~~ (done), so the only remaining systems would be explosives and disposable, which affect only a minimum fraction of third party content (e.g., doesn't affect vehicles nor small arms).

The current fix is fast (the extension takes 0.01 ms to calculate, and only is required to do so once every time the unit changes zeroing, weapon or ammo) and seems to be as accurate. Fortunately `currentZeroing` works properly with remote units.

**TODO** 
- [ ] Hide the vanilla zeroing control when appropiate, so the end user won't even know it's still enabled. Tagging @jonpas for advice
- [ ] Restore `discreteDistance[] = {}` for vanilla content?
- [ ] Correctly compute the initSpeed from config instead of getting it from the bullet?

### CODE REVIEW

**Comment by jonpas** [comment]
> Hide the vanilla zeroing control when appropiate, so the end user won't even know it's still enabled.

I will add a public function for hiding UI parts, will also act like a setter so UI component itself won't overwrite it (as it has to be set on vehicle changes as well).

**Comment by commy2** [comment]
This would be excellent! :+1:

**Comment by esteldunedain** [comment]
I added a first implementation of the correction to ACE_Scopes. I didn't test this yet, so it might still contain bugs.

**Comment by Khaosmatic** [comment]
Regarding vehicle FCS, how is ACE's going to work alongside BI's 1.60 enhancements?

**Comment by ulteq** [comment]
> I added a first implementation of the correction to ACE_Scopes.

Won't this cause all scopes to be zeroed at 0 meters?

**Comment by ulteq** [comment]
> The current fix is fast (the extension takes 0.01 ms to calculate, and only is required to do so once every time the unit changes zeroing, weapon or ammo) and seems to be as accurate. Fortunately `currentZeroing` works properly with remote units.
How about passing the `currentZeroing` value to the FCS extension and let it handle everything?

File: addons/scopes/functions/fnc_firedEH.sqf [hunk 1]
```diff
@@ -15,20 +15,49 @@
...
 _zeroing params ["_elevation", "_windage", "_zero"];
-
-[_projectile, _windage, _elevation + _zero, 0] call EFUNC(common,changeProjectileDirection);
+_elevation = _elevation + _zero;
+
+// The scope supports vertical adjustment, so even if _zeroing is [0,0,0] we
+// still need to compute the correction to compensate for vanilla zeroing
+if (_adjustVertical) then {
+    // Calculate the correction due to vanilla zeroing
+    private _zeroDistance = (currentZeroing _unit - GVAR(defaultZeroDistance));
```
Comment (ulteq) [inline addons/scopes/functions/fnc_firedEH.sqf]: I'm pretty sure that this won't work well.

**Comment by esteldunedain** [comment]
Closing in favor of #4642 and #4653

**Pull request closed without merging**